src/utils/analytics.ts

```diff
@@ -131,69 +137,220 @@ export enum PageType {
+
+/**
+ * Track page views
+ */
+export const trackPage = async (title: PageType) => {
   try {
-    await analytics.page("Wander Extension", {
-      title,
-    });
+    // Only track BE production events (unless dev analytics is explicitly enabled):
+    if (
+      (process.env.NODE_ENV === "development" && !ENABLE_DEV_ANALYTICS) ||
+      import.meta.env?.VITE_IS_EMBEDDED_APP === "1"
+    ) {
+      return;
+    }
+
+    const enabled = await getSetting("analytics").getValue();
+    if (!enabled) return;
+
+    await analytics.page({ title });
   } catch (err) {
-    console.log("err", err);
+    log(LOG_GROUP.ANALYTICS, "Page tracking error:", err);
   }
 };
 
+/**
+ * Track events directly (for background scripts)
+ */
 export const trackDirect = async (event: EventType, properties: Record<string, unknown>) => {
-  // Only track BE production events:
-  if (process.env.NODE_ENV === "development" || import.meta.env?.VITE_IS_EMBEDDED_APP === "1") return;
-
-  const enabled = await getSetting("analytics").getValue();
+  try {
+    // Only track BE production events (unless dev analytics is explicitly enabled):
+    if (
+      (process.env.NODE_ENV === "development" && !ENABLE_DEV_ANALYTICS) ||
+      import.meta.env?.VITE_IS_EMBEDDED_APP === "1"
+    ) {
+      return;
+    }
 
-  if (!enabled) return;
+    const enabled = await getSetting("analytics").getValue();
+    if (!enabled) return;
 
-  // TODO: We probably want to update this for Connect:
-  let userId = await ExtensionStorage.get("user_id");
-  if (!userId) {
-    userId = nanoid();
-    await ExtensionStorage.set("user_id", userId);
+    await analytics.track(event, { ...properties });
+  } catch (err) {
+    log(LOG_GROUP.ANALYTICS, `Failed to track event ${event}:`, err);
   }
-
-  return fetch("https://api.segment.io/v1/t", {
-    method: "POST",
-    body: JSON.stringify({
-      event,
-      properties,
-      messageId: nanoid(),
-      sentAt: new Date().toISOString(),
-      timestamp: new Date().toISOString(),
-      type: "track",
-      userId: userId,
-      writeKey: PUBLIC_SEGMENT_WRITEKEY,
-    }),
-  });
 };
 
+/**
+ * Track custom events
+ */
 export const trackEvent = async (eventName: EventType, properties: any) => {
   try {
-    // Only track BE production events:
-    if (process.env.NODE_ENV === "development" || import.meta.env?.VITE_IS_EMBEDDED_APP === "1") return;
+    // Only track BE production events (unless dev analytics is explicitly enabled):
+    if (
+      (process.env.NODE_ENV === "development" && !ENABLE_DEV_ANALYTICS) ||
+      import.meta.env?.VITE_IS_EMBEDDED_APP === "1"
+    ) {
+      return;
+    }
 
     // first we check if we are allowed to collect data
     const enabled = await getSetting("analytics").getValue();
-
     if (!enabled) return;
 
     const ONE_HOUR_IN_MS = 3600000;
 
     // TODO:login is tracked only once and compared to an hour period before logged as another Login event
     if (eventName === EventType.LOGIN) {
```

Comment:
@7i7o should we use local or session storage for login event as currently we use session and the stored data is lost when browser is closed ? wdyt ?